tools/launch.py

```diff
@@ -105,10 +108,17 @@ def execute_remote(
     )
 
     # thread func to run the job
-    def run(ssh_cmd):
-        subprocess.check_call(ssh_cmd, shell=True)
-
-    thread = Thread(target=run, args=(ssh_cmd,))
+    def run(ssh_cmd, state_q):
+        try:
+            subprocess.check_call(ssh_cmd, shell=True)
+        except subprocess.CalledProcessError as err:
+            print(f"Called process error {err}")
+            state_q.put(err.returncode)
+        except Exception:
+            state_q.put(-1)
+        state_q.put(0)
```

Comment:
i think you should move this line to after line 113.

Reply:
otherwise, each run adds two values in the queue.

Reply:
Done.
You are right.